apps/server/src/services/rundown-service/rundownCache.ts

```diff
@@ -112,11 +115,18 @@ export function generate(
           firstStart = currentEntry.timeStart;
         }
 
+        const isNextDay =
+          lastEntry === null ? 0 : checkIsNextDay(lastEntry.timeStart, currentEntry.timeStart, lastEntry.duration);
+
+        dayOffset += isNextDay ? 1 : 0;
+        currentEntry.dayOffset = dayOffset;
+
         const timeFromPrevious: number = getTimeFromPrevious(
           currentEntry.timeStart,
+          currentEntry.dayOffset,
```

Comment:
cleanup: we need to make sure that these properties do not get persisted. I have flagged the entrypoint for the code in this comment
https://github.com/cpvalente/ontime/blob/master/apps/server/src/services/rundown-service/rundownCache.ts#L97